packages/mask/src/plugins/Referral/helpers/index.tsx

```diff
@@ -0,0 +1,79 @@
+import { padStart } from 'lodash-unified'
+import BigNumber from 'bignumber.js'
+import { createTypedMessageMetadataReader } from '@masknet/typed-message/dom'
+import { formatUnits } from '@ethersproject/units'
+
+import { META_KEY, supportedChainId } from '../constants'
+import type {
+    ReferralMetaData,
+    ChainAddress,
+    ChainAddressProps,
+    EvmAddress,
+    Bytes32,
+    Bytes24,
+    ChainId,
+    BigNumberish,
+} from '../types'
+import schema from '../schema.json'
+import { bufToHexString, toBigInt, writeUInt32BE, hexToArrayBuffer } from './buffer'
+
+function toChainAddress(chainId: BigNumber | bigint, address: Uint8Array): Uint8Array {
+    if (address.byteLength !== 20) throw new Error('invalid address')
```

Comment:
Learn more: `packages/web3-shared/evm/utils/address.ts`.

```suggestion
    if (isValidAddress(address)) throw new Error('invalid address')
```

Reply:
Done